core/go/internal/publictxmgr/in_flight_transaction_state_version.go

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright © 2025 Kaleido, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package publictxmgr
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kaleido-io/paladin/core/internal/msgs"
+	"github.com/kaleido-io/paladin/core/pkg/ethclient"
+	"github.com/kaleido-io/paladin/toolkit/pkg/i18n"
+	"github.com/kaleido-io/paladin/toolkit/pkg/log"
+	"github.com/kaleido-io/paladin/toolkit/pkg/pldapi"
+	"github.com/kaleido-io/paladin/toolkit/pkg/tktypes"
+)
+
+type inFlightTransactionStateVersion struct {
+	// This id is never persisted in the database it only exists in memory so that we can pass around a reference
+	// to the version id in some functions rather than passing the reference to the version struct itself. This
+	// is because having any types imported from this package into mocks files results in circular dependencies
+	id int
+
+	current             bool
+	testOnlyNoEventMode bool
+
+	PublicTxManagerMetricsManager
+	InFlightStageActionTriggers
+	InMemoryTxStateManager
+
+	// input that should be set once the stage is running
+	*TransientPreviousStageOutputs
+
+	// the current in-flight stage
+	// this is the core of in-flight transaction processing.
+	// only 1 stage context can exist at any given time for a specific transaction version.
+	// in the case there are multiple versions, only the current version is allowed to progress,
+	// previous versions may persist what they have done so far but no more
+	// in flight transaction contains the logic to process each stage to its completion,
+	// any stage will have at least 1 asynchronous action, in-flight transaction relies on transaction orchestrator
+	// to give it signal to collect result of those async actions.
+	// Therefore, any coordination required cross in-flight transaction can be taken into consideration for next stage.
+	//    e.g. even if the transaction is ready for submission, we might not want to submit it if the other transactions
+	//     ahead of the current transaction used up all the funds
```

Comment:
This is a gem of a comment.

```
TX versions (many) -> current version (singular) -> running stage context (singular) -> action (0/1)
```

Reply:
It doesn't tell me what is above TX Version in the hierachy, as I know from previous experience with the code that many, many TXs (each managing their own version) might be being managed on a signing address at any given time. But this comment feels like painting the start of the picture that would help someone have a clear reference for what this code is and how it's structured 🙌